### PR TITLE
clash-rs: init at 0.7.0

### PR DESCRIPTION
**Pull request by ghost**
ClashRS, A custom protocol, rule based network proxy software. https://github.com/Watfaq/clash-rs
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/cl/clash-rs/package.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  version = "0.7.0";
+
+  src = fetchFromGitHub {
+    owner = "Watfaq";
+    repo = "clash-rs";
+    rev = "v${version}";
+    hash = "sha256-0deMVI51XHTCrnLTycqDsaY5Lq+wx14uMUlkG5OViNA=";
+  };
+
+  RUSTC_BOOTSTRAP = 1;
```
Comment (Aleksanaa) [inline pkgs/by-name/cl/clash-rs/package.nix:18]: Why not combine it with the env below?

File: pkgs/by-name/cl/clash-rs/package.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      "tracing-oslog-0.2.0" = "sha256-JYaCslbVOgsyBhjeBkplPWcjSgFccjr4s6OAGIUu5kg=";
+      "tuic-1.3.1" = "sha256-WMd+O2UEu0AEI+gNeQtdBhEgIB8LPanoIpMcDAUUWrM=";
+      "tun-0.6.1" = "sha256-j4yQSu4Mw7DBFak8vJGQomYq81+pfaeEDdN4NNBve+E=";
+      "unix-udp-sock-0.7.0" = "sha256-TekBfaxecFPpOfq7PVjLHwc0uIp3yJGV/Cgav5VfKaA=";
+    };
+  };
+
+  env.PROTOC = "${protobuf}/bin/protoc";
+
+  doCheck = false;
```
Comment (Aleksanaa) [inline pkgs/by-name/cl/clash-rs/package.nix:36]: Add reason in comment near it

File: pkgs/by-name/cl/clash-rs/package.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  meta = {
+    description = "custom protocol network proxy";
```
Comment (Aleksanaa) [inline pkgs/by-name/cl/clash-rs/package.nix:39]:
```suggestion
    description = "Custom protocol, rule based network proxy software";
```

**Comment by Aleksanaa** [comment]
Why are you targeting staging-next? Is there something truly needed?

File: pkgs/by-name/cl/clash-rs/package.nix (hunk 2 above)
Comment (Aleksanaa) [reply pkgs/by-name/cl/clash-rs/package.nix:34]: Annotation 🙃

**Review by zi3m5f — APPROVED** [review]
LGTM :+1:

- [x] builds as is
- [x] builds with `versionCheckHook` additions
- [x] builds with extra features, see below

I noticed `-F "shadowsocks, tuic, onion"` in the github ci workflow building the official binaries:
https://github.com/Watfaq/clash-rs/blob/98d183b04f96b8bcabe265af849713abeb5f40c3/.github/workflows/ci.yml#L48

Maybe add them to not differ from upstream?
```nix
  buildFeatures = [ "shadowsocks" "tuic" "onion" ];
```
seems to work:
> cargoBuildHook flags: -j 12 --target x86_64-unknown-linux-gnu --offline --profile release --features=shadowsocks\,tuic\,onion

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**